codex/node.nim

```diff
@@ -514,6 +514,32 @@ proc requestStorage*(
   let purchase = await contracts.purchasing.purchase(request)
   success purchase.id
 
+proc debugDelete*(self: CodexNodeRef, cid: Cid, blockIndices: seq[int]): Future[?!int] {.async.} =
+  without manifest =? (await self.fetchManifest(cid)), err:
+    trace "Unable to fetch manifest for cid", cid, err = err.msg
+    return failure(err)
+
+  let
+    totalBlocks = manifest.blocksCount
+    treeCid = manifest.treeCid
+
+  var
+    deleted = 0
+
+  for index in blockIndices:
+    if index >= totalBlocks:
+      return failure("Index out of range. Index: " & $index & " totalBlocks: " & $totalBlocks)
+    without hasBlock =? (await self.networkStore.hasBlock(treeCid, index)), err:
+      return failure(err)
+    if not hasBlock:
+      return failure("Tree does not have this index: " & $index & " totalBlocks: " & $totalBlocks)
+    else:
+      if err =? (await self.networkStore.delBlock(treeCid, index)).errorOption:
+        return failure(err)
+      inc deleted
```

Comment:
Why count the deletes if you only return success when all deletes succeed?

Reply:
It's not strictly necessary. If it's success, you know it should have worked. I was thinking the test-code should assert the number that is returned here. Just to have an extra check on the state of the node. What do you think?

Reply:
Hm... you mean to assert that the delete code is doing what it's supposed to? I mean, I can live with that if you think it'll make you feel safer using it, but this means you're delegating testing the API to the client in some sense. For a debugging API could be fine.

Reply:
Well yes, the whole thing is behind a compile flag. it's really only there for testing.
When I call this thing with X indices, I will always expect an error OR exactly X in the response. I don't think this will ever behave differently. *BUT* in case it does, in case some code gets changed in the future, in case the laws of physics no longer apply... I *REALLY* want to know. Please permit me this paranoia. :D